src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -1153,6 +1152,23 @@ REDHAWK_PALEXPORT int32_t PalGetProcessCpuCount()
     return g_RhNumberOfProcessors;
 }
 
+REDHAWK_PALEXPORT uint32_t REDHAWK_PALAPI PalGetOsPageSize()
+{
+    static int saved_pagesize = 0;
+
+    if (saved_pagesize)
+        return saved_pagesize;
+
+    // Prefer sysconf () as it's signal safe.
```

Comment:
Do we care about signal safety here? If we do, we can initialize this during `PalInit` like it is done in CoreCLR.

Reply:
The same pattern is used in mono in multiple paces. I've just copied it, together with the comment. I guess when either one can be used signal safety might be an advantage.  
We do not call this from code that must be signal-safe right now.

Reply:
I think we can drop the comment about signal safety - not very relevant to us, but keep the pattern - use `sysconf`, if can, then use `getpagesize`.